README.md
# Search Operator Bundle

Bundles the latest Odyssey (OCM Search v2) components.

## Image Pickup

> Pre-requisites:
>
> - You will need a `GITHUB_TOKEN` that provides read access from the [stolostron/pipeline](https://github.com/stolostron/pipeline) (private) repository (required to fetch the latest image manifests).
> - jq command - [link](https://formulae.brew.sh/formula/jq)
> - yq command - [link](https://formulae.brew.sh/formula/yq)
> - When running the update manually, you will need to ensure that the `snapshot.ver` and `pg-snapshot.ver` file exist within the `search-v2-operator-bundle` repository.

When bundling the latest build, the following `Search` images will be targeted (unless ignored):

- [search-collector](https://github.com/stolostron/search-collector)
- [search-indexer](https://github.com/stolostron/search-indexer)
- [search-v2-api](https://github.com/stolostron/search-v2-api)
- [search-v2-operator](https://github.com/stolostron/search-v2-operator)

The following `Postgresql` image will be targeted (unless ignored):

- [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362)

### Running the bundle update locally

When executing the script, you will be prompted to select the `SNAPSHOT` tag for the `Search` and `Postgresql` images. By default, this project support the usage of the `snapshot.ver` and `pg-snapshot.ver` file.

The default tag version for the `Search` components will be located within the `snapshot.ver`. To verify that the file exist, run the following command:

```bash
# 2.7.0-SNAPSHOT-2022-09-15-08-33-34
cat ./snapshot.ver
```

The default tag version for the `Postgresql` component will be located within the `pg-snapshot.ver`. To verify that the file exist, run the following command:

```bash
# 1-78
cat ./pg-snapshot.ver
```

To continue with the default tags that are set within the repository, simply press `enter` or `return`; otherwise, you can fetch the required `SNAPSHOT` tag to use from the [quay.io/stolostron](https://quay.io/organization/stolostron) registry.

![Manual Update](docs/images/bundle-image-pickup-manual.gif)

**NOTE**: The default image tags that are being used within the `snapshot.ver` and `pg-snapshot.ver` will only be updated after running the script `manually`. If the user execute the script while silent mode is enabled, the `snapshot` files will not be updated with the latest image tags.

### Running the bundle update in silent mode

For automatic updates, the user can run the script in `silent` mode. When running the script within `silent` mode, the user will need to ensure that they have access to the pipeline repository, from which the latest images will be fetched from. The user will also need to export their github token as the following environment variable: `GITHUB_TOKEN`.

To run the script within silent mode, add the `--silent` or `-s` flag at the end of the command to ensure that the mode is enabled.

```bash
./scripts/bundle-image-pickup.sh --silent

or 

./scripts/bundle-image-pickup.sh -s
```

![Automated Update](docs/images/bundle-image-pickup-silent.gif)

### TODO: Scheduling bundle update with Github actions workflow

Details to be added...

### Environment Variables

Each environment variable is customizable and can be set by the user.

| Name                          | Description                                                        | Default Value                                               |
|-------------------------------|--------------------------------------------------------------------|-------------------------------------------------------------|
| IMG_REGISTRY                  | Image registry from where the build image will be pulled from      | quay.io/$ORG                                                |
| IGNORE_API_IMAGE_UPDATE       | Ignore search-v2-api image update                                  | false                                                       |
| IGNORE_COLLECTOR_IMAGE_UPDATE | Ignore search-collector image update                               | false                                                       |
| IGNORE_INDEXER_IMAGE_UPDATE   | Ignore search-indexer image update                                 | false                                                       |
| IGNORE_OPERATOR_IMAGE_UPDATE  | Ignore search-v2-operator image update                             | false                                                       |
| IGNORE_POSTGRES_IMAGE_UPDATE  | Ignore postgresql-13 image update                                  | true                                                        |
| OPERATOR_CONTAINER_PATH       | The container path within the search-v2-operator.clusterserviceversion manifest file | .spec.install.spec.deployments[0].spec.template.spec.containers[1] |
| OPERATOR_CSV_FILEPATH         | The search-v2-operator.clusterserviceversion manifest file path | bundle/manifests/search-v2-operator.clusterserviceversion.yaml |
| OPERATOR_ENV_PATH             | The container env path within the search-v2-operator.clusterserviceversion manifest file   | OPERATOR_CONTAINER_PATH.env[].value |
| OPERATOR_IMAGE_PATH           | The container image path within the search-v2-operator.clusterserviceversion manifest file | OPERATOR_CONTAINER_PATH.image       |
| ORG                           | The org from which the images will be pulled from and where the manifest file will be fetched from | stolostron                  |
| PIPELINE_REPO                 | The repository that contains the image manifest file                     | pipeline                                              |
| POSTGRES_IMG_REGISTRY         | The image registry for the postgresql component                          | $IMG_REGISTRY                                         |
| README_FILEPATH               | The README filepath for where the updated status will be added           | README.md                                             |
| RELEASE_BRANCH                | The release branch from which the pipeline manifest will be fetched from | 2.7-integration                                       |
| SEARCH_API_IMG_REGISTRY       | The image registry for the search-v2-api component                       | $IMG_REGISTRY                                         |
| SEARCH_COLLECTOR_IMG_REGISTRY | The image registry for the search-collector component                    | $IMG_REGISTRY                                         |
| SEARCH_INDEXER_IMG_REGISTRY   | The image registry for the search-indexer component                      | $IMG_REGISTRY                                         |
| SEARCH_OPERATOR_IMG_REGISTRY  | The image registry for the search-v2-operator component                  | $IMG_REGISTRY                                         |

## Links

These are a few useful links that will help provide technical reference and best practices when developing for the platform.

- [jq Docs](https://stedolan.github.io/jq/tutorial/)
- [yq Docs](https://mikefarah.gitbook.io/yq/)

Image Rebuild: Thu Sep 15 13:21:16 EDT 2022


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## Bundle Image Update Logs

### Date of Change: 08-23-2022

---

#### Updated Build Version: Tue Aug 23 19:03:34 EDT 2022

| Image Name                                                             | Image Component  |
|------------------------------------------------------------------------|------------------|
| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | registry.redhat.io/rhel8/postgresql-13:1-56  |
| [search-collector](https://github.com/stolostron/search-collector)     | quay.io/stolostron/search-collector:2.7.0-5f707444e4187a6dc72edd891a2ff97ffc3e6ca8 |
| [search-indexer](https://github.com/stolostron/search-indexer)         | quay.io/stolostron/search-indexer:2.7.0-4c1b1f4f692d5f8be9fca699ea26573227250f39   |
| [search-v2-api](https://github.com/stolostron/search-v2-api)           | quay.io/stolostron/search-v2-api:2.7.0-0da7af171e5cc5c1b319cc9e7710d13db85734d1       |
| [search-v2-operator](https://github.com/stolostron/search-v2-operator) | quay.io/stolostron/search-v2-operator:2.7.0-8220bc29e9b609b1f27a4a221962ebeb57dc25cf  |

### Date of Change: 08-29-2022

---

#### Updated Build Version: Mon Aug 29 19:31:31 EDT 2022

| Image Name                                                             | Image Component  |
|------------------------------------------------------------------------|------------------|
| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | registry.redhat.io/rhel8/postgresql-13:1-56  |
| [search-collector](https://github.com/stolostron/search-collector)     | quay.io/stolostron/search-collector:2.6.0-SNAPSHOT-2022-08-08-20-23-21 |
| [search-indexer](https://github.com/stolostron/search-indexer)         | quay.io/stolostron/search-indexer:2.6.0-SNAPSHOT-2022-08-08-20-23-21   |
| [search-v2-api](https://github.com/stolostron/search-v2-api)           | quay.io/stolostron/search-v2-api:2.6.0-SNAPSHOT-2022-08-08-20-23-21       |
| [search-v2-operator](https://github.com/stolostron/search-v2-operator) | quay.io/stolostron/search-v2-operator:2.6.0-SNAPSHOT-2022-08-08-20-23-21  |

#### Updated Build Version: Mon Aug 29 19:32:01 EDT 2022

| Image Name                                                             | Image Component  |
|------------------------------------------------------------------------|------------------|
| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | registry.redhat.io/rhel8/postgresql-13:1-56  |
| [search-collector](https://github.com/stolostron/search-collector)     | quay.io/stolostron/search-collector:2.7.0-de589fdba4effc06198db8250c1ae3118125d685 |
| [search-indexer](https://github.com/stolostron/search-indexer)         | quay.io/stolostron/search-indexer:2.7.0-4c1b1f4f692d5f8be9fca699ea26573227250f39   |
| [search-v2-api](https://github.com/stolostron/search-v2-api)           | quay.io/stolostron/search-v2-api:2.7.0-69590634c2de51ceec786af47ca5fb00f42a631a       |
| [search-v2-operator](https://github.com/stolostron/search-v2-operator) | quay.io/stolostron/search-v2-operator:2.7.0-dc0fc8c59239b6a0a3f440b7752101523d87f597  |

### Date of Change: 09-01-2022

---

#### Updated Build Version: Thu Sep  1 16:13:08 EDT 2022

| Image Name                                                             | Image Component  |
|------------------------------------------------------------------------|------------------|
| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | registry.redhat.io/rhel8/postgresql-13:1-56  |
| [search-collector](https://github.com/stolostron/search-collector)     | quay.io/stolostron/search-collector: |
| [search-indexer](https://github.com/stolostron/search-indexer)         | quay.io/stolostron/search-indexer:   |
| [search-v2-api](https://github.com/stolostron/search-v2-api)           | quay.io/stolostron/search-v2-api:       |
| [search-v2-operator](https://github.com/stolostron/search-v2-operator) | quay.io/stolostron/search-v2-operator:  |

#### Updated Build Version: Thu Sep  1 16:15:09 EDT 2022

| Image Name                                                             | Image Component  |
|------------------------------------------------------------------------|------------------|
| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | registry.redhat.io/rhel8/postgresql-13:1-56  |
| [search-collector](https://github.com/stolostron/search-collector)     | quay.io/stolostron/search-collector: |
| [search-indexer](https://github.com/stolostron/search-indexer)         | quay.io/stolostron/search-indexer:   |
| [search-v2-api](https://github.com/stolostron/search-v2-api)           | quay.io/stolostron/search-v2-api:       |
| [search-v2-operator](https://github.com/stolostron/search-v2-operator) | quay.io/stolostron/search-v2-operator:  |

### Date of Change: 09-02-2022

---

#### Updated Build Version: Fri Sep  2 07:55:48 EDT 2022

| Image Name                                                             | Image Component  |
|------------------------------------------------------------------------|------------------|
| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | registry.redhat.io/rhel8/postgresql-13:1-56  |
| [search-collector](https://github.com/stolostron/search-collector)     | quay.io/stolostron/search-collector:2.7.0-de589fdba4effc06198db8250c1ae3118125d685 |
| [search-indexer](https://github.com/stolostron/search-indexer)         | quay.io/stolostron/search-indexer:2.7.0-4c1b1f4f692d5f8be9fca699ea26573227250f39   |
| [search-v2-api](https://github.com/stolostron/search-v2-api)           | quay.io/stolostron/search-v2-api:2.7.0-260947980e11381931b7d193370baa3596beed7f       |

scripts/bundle-image-pickup.sh
#!/bin/bash

###############################################################################
# Copyright Contributors to the Open Cluster Management project
###############################################################################

source ./scripts/log-colors.sh

log_color "cyan" "Initializing search bundle image pickup..."
echo -e "Current dir: $(pwd)\n"

####################
## ENV VARIABLES
####################
ORG=${ORG:-"stolostron"}
PIPELINE_REPO=${PIPELINE_REPO:-"pipeline"}
RELEASE_BRANCH=${RELEASE_BRANCH:-"2.7-integration"}

IMG_REGISTRY=${IMG_REGISTRY:-"quay.io/$ORG"}
SEARCH_API_IMG_REGISTRY=${SEARCH_API_IMG_REGISTRY:-$IMG_REGISTRY}
SEARCH_COLLECTOR_IMG_REGISTRY=${SEARCH_COLLECTOR_IMG_REGISTRY:-$IMG_REGISTRY}
SEARCH_INDEXER_IMG_REGISTRY=${SEARCH_INDEXER_IMG_REGISTRY:-$IMG_REGISTRY}
SEARCH_OPERATOR_IMG_REGISTRY=${SEARCH_OPERATOR_IMG_REGISTRY:-$IMG_REGISTRY}
POSTGRES_IMG_REGISTRY=${POSTGRES_IMG_REGISTRY:-$IMG_REGISTRY}

####################
## IGNORE VARIABLES
####################
IGNORE_API_IMAGE_UPDATE=${IGNORE_API_IMAGE_UPDATE:-"false"}
IGNORE_COLLECTOR_IMAGE_UPDATE=${IGNORE_COLLECTOR_IMAGE_UPDATE:-"false"}
IGNORE_INDEXER_IMAGE_UPDATE=${IGNORE_INDEXER_IMAGE_UPDATE:-"false"}
IGNORE_OPERATOR_IMAGE_UPDATE=${IGNORE_OPERATOR_IMAGE_UPDATE:-"false"}
IGNORE_POSTGRES_IMAGE_UPDATE=${IGNORE_POSTGRES_IMAGE_UPDATE:-"true"}

####################
## PATHS (I.E DIR, FILES, ETC)
####################
OPERATOR_CSV_FILEPATH=${OPERATOR_CSV_FILEPATH:-"bundle/manifests/search-v2-operator.clusterserviceversion.yaml"}
CHANGELOG_FILEPATH=${CHANGELOG_FILEPATH:-"CHANGELOG.md"}

OPERATOR_CONTAINER_PATH=${OPERATOR_CONTAINER_PATH:-".spec.install.spec.deployments[0].spec.template.spec.containers[1]"}
OPERATOR_ENV_PATH=${OPERATOR_ENV_PATH:-"$OPERATOR_CONTAINER_PATH.env[].value"}
OPERATOR_IMAGE_PATH=${OPERATOR_IMAGE_PATH:-"$OPERATOR_CONTAINER_PATH.image"}

####################
## FUNCTIONS/METHODS
####################
display_component_images () {
  echo -e "Component Images"
  echo -e "==============================================================================" \
  "\nPOSTGRES:\t\t${POSTGRES_IMAGE}" \
  "\nSEARCH_API:\t\t${API_IMAGE}" \
  "\nSEARCH_COLLECTOR:\t${COLLECTOR_IMAGE}" \
  "\nSEARCH_INDEXER:\t\t${INDEXER_IMAGE}" \
  "\nSEARCH_OPERATOR:\t${OPERATOR_IMAGE}" \
  "\n==============================================================================\n"
}

ignore_component_update() {
  COMPONENT=$1
  case $COMPONENT in
    postgresql-13)
      IGNORE=$IGNORE_POSTGRES_IMAGE_UPDATE
    ;;
    search-collector)
      IGNORE=$IGNORE_COLLECTOR_IMAGE_UPDATE
    ;;
    search-indexer)
      IGNORE=$IGNORE_INDEXER_IMAGE_UPDATE
    ;;
    search-v2-api)
      IGNORE=$IGNORE_API_IMAGE_UPDATE
    ;;
    search-v2-operator)
      IGNORE=$IGNORE_OPERATOR_IMAGE_UPDATE
    ;;
    *)
      IGNORE=true
    ;;
  esac
  echo $IGNORE
}

get_images_from_csv () {
  log_color purple "Fetching component images from ${OPERATOR_CSV_FILEPATH}" "\n"

  for IMG in $(yq e ${OPERATOR_IMAGE_PATH} ${OPERATOR_CSV_FILEPATH}); do
    if [[ $IMG =~ .*"search-v2-operator".* ]]; then
      OPERATOR_IMAGE=$IMG
    fi
  done

  for IMG in $(yq e ${OPERATOR_ENV_PATH} ${OPERATOR_CSV_FILEPATH}); do
    if [[ $IMG =~ .*"postgres".* ]]; then
      POSTGRES_IMAGE=$IMG

    elif [[ $IMG =~ .*"search-collector".* ]]; then
      COLLECTOR_IMAGE=$IMG

    elif [[ $IMG =~ .*"search-indexer".* ]]; then
      INDEXER_IMAGE=$IMG

    elif [[ $IMG =~ .*"search-v2-api".* ]]; then
      API_IMAGE=$IMG
    fi
  done

  display_component_images
}

update_doc_entry () {
  # Check to see if the current date header is within the bundle-image-update.md file.
  if ! grep -q $(date +%m-%d-%Y) $CHANGELOG_FILEPATH; then
        echo -e "\n### Date of Change: $(date +%m-%d-%Y)\n" \
    "\n---" >> $CHANGELOG_FILEPATH
  fi

  echo -e "\n#### Updated Build Version: $(date)" >> $CHANGELOG_FILEPATH
  echo -e "\n| Image Name                                                             | Image Component  |\n" \
      "|------------------------------------------------------------------------|------------------|\n" \
      "| [postgresql-13](https://catalog.redhat.com/software/containers/rhel8/postgresql-13/5ffdbdef73a65398111b8362) | ${POSTGRES_IMAGE}  |\n" \
      "| [search-collector](https://github.com/stolostron/search-collector)     | ${COLLECTOR_IMAGE} |\n" \
      "| [search-indexer](https://github.com/stolostron/search-indexer)         | ${INDEXER_IMAGE}   |\n" \
      "| [search-v2-api](https://github.com/stolostron/search-v2-api)           | ${API_IMAGE}       |\n" \
      "| [search-v2-operator](https://github.com/stolostron/search-v2-operator) | ${OPERATOR_IMAGE}  |" >> $CHANGELOG_FILEPATH

  sed -i'.bak' 's/^[ \t]*//'  $CHANGELOG_FILEPATH
}

update_images_csv () {
  COMPONENT=$1
  NEW_IMAGE=$2
  IGNORE=$3

  log_color purple "Preparing to update component: ${COMPONENT} => ${NEW_IMAGE}" "\n"

   if [[ $COMPONENT =~ .*"postgresql-13".* ]]; then
    yq -i e "${OPERATOR_CONTAINER_PATH}.env[1].value = \"${NEW_IMAGE}\"" $OPERATOR_CSV_FILEPATH
  
  elif [[ $COMPONENT =~ .*"search-indexer".* ]]; then
    yq -i e "${OPERATOR_CONTAINER_PATH}.env[2].value = \"${NEW_IMAGE}\"" $OPERATOR_CSV_FILEPATH

  elif [[ $COMPONENT =~ .*"search-collector".* ]]; then
    yq -i e "${OPERATOR_CONTAINER_PATH}.env[3].value = \"${NEW_IMAGE}\"" $OPERATOR_CSV_FILEPATH
  
  elif [[ $COMPONENT =~ .*"search-v2-api".* ]]; then
    yq -i e "${OPERATOR_CONTAINER_PATH}.env[4].value = \"${NEW_IMAGE}\"" $OPERATOR_CSV_FILEPATH

  elif [[ $COMPONENT =~ .*"search-v2-operator".* && $IGNORE_POSTGRES_IMAGE_UPDATE == true ]]; then
    yq -i e "${OPERATOR_IMAGE_PATH} = \"${NEW_IMAGE}\"" $OPERATOR_CSV_FILEPATH
  fi
}

# To run in silent mode the user's GITHUB_TOKEN will need to be exported. This is needed to fetch the pipeline manifest.json file from the stolostron org.
if [[ " $@ " =~ " --silent " && -z $GITHUB_TOKEN || " $@ " =~ " -s " && -z $GITHUB_TOKEN ]]; then
  log_color "yellow" "\"GITHUB_TOKEN\" is not exported; (\"GITHUB_TOKEN\" is required to execute the script in silent mode)" 
  exit 1
fi

# Fetch the current images from the csv file.
get_images_from_csv

# Set the default snapshot version.
if [[ -f ./snapshot.ver && -f ./pg-snapshot.ver ]]; then
  DEFAULT_SNAPSHOT=`cat ./snapshot.ver`
  DEFAULT_POSTGRESQL_SNAPSHOT=`cat ./pg-snapshot.ver`

elif [[ " $@ " =~ " --silent " || " $@ " =~ " -s " ]]; then
  log_color "red" "ERROR: Silent mode will not work when ./snapshot.ver or ./pg-snapshot.ver is missing"
  exit 1
fi

# Input or fetch the latest build image to update the csv file.
if [[ " $@ " =~ " --silent " || " $@ " =~ " -s " ]]; then
  log_color "cyan" "** Running in silent mode **"
else
  echo -e "Find snapshot tags @ https://quay.io/repository/stolostron/acm-custom-registry?tab=tags\nEnter SNAPSHOT TAG (SEARCH): (Press ENTER for default: ${DEFAULT_SNAPSHOT})\n"
  read -r SNAPSHOT_CHOICE

  if [[ "${SNAPSHOT_CHOICE}" != "" ]]; then
    DEFAULT_SNAPSHOT=$SNAPSHOT_CHOICE
    echo -e $DEFAULT_SNAPSHOT > ./snapshot.ver
  fi

  echo -e "Find snapshot tags @ https://catalog.redhat.com/software/containers/search?q=postgresql&p=1\nEnter SNAPSHOT TAG (POSTGRESQL): (Press ENTER for default: ${DEFAULT_POSTGRESQL_SNAPSHOT})\n"
  read -r SNAPSHOT_CHOICE

  if [[ "${SNAPSHOT_CHOICE}" != "" ]]; then
    DEFAULT_POSTGRES_SNAPSHOT=$SNAPSHOT_CHOICE
    echo -e $DEFAULT_SNAPSHOT > ./pg-snapshot.ver
  fi
fi

# Create an array containing the Search components that we will focus on for image versioning.
SEARCH_COMPONENTS=(postgresql-13 search-collector search-indexer search-v2-api search-v2-operator)

# Fetch component images and add current entry to bundle history markdown file.
# Running in silent mode will automatically fetch the latest image tag available within the stolostron/pipeline manifest.
if [[ " $@ " =~ " --silent " || " $@ " =~ " -s " ]]; then
  # Get the latest manifest file to capture the latest builds.
  PIPELINE_MANIFEST=$(curl GET https://raw.githubusercontent.com/$ORG/$PIPELINE_REPO/$RELEASE_BRANCH/manifest.json -H "Authorization: token $GITHUB_TOKEN")
  log_color "purple" "\nFetching image-tags from pipeline manifest." "\n"

  # Check to see if the pipeline manifest file contains the image-tag. If not, then there is an error or issue with the file.
  if [[ ! $PIPELINE_MANIFEST =~ "image-tag" ]]; then
    log_color "red" "[ERROR] PIPELINE_MANIFEST does not contain \"image-tag\". Exiting the script."
    exit 1
  fi
fi

for COMPONENT in ${SEARCH_COMPONENTS[@]}; do
  log_color "yellow" "Component: $COMPONENT"

  # Check to see if component update needs to be ignored.
  IGNORE=$(ignore_component_update $COMPONENT)

  if [[ $IGNORE == true ]]; then
    echo -e "Skipping image build update for component: $COMPONENT (IGNORE set to true)\n"
  else
    # Generate the base image.
    IMAGE=$IMG_REGISTRY/$COMPONENT

    if [[ " $@ " =~ " --silent " || " $@ " =~ " -s " ]]; then
      # Fetch search component within the manifest file.
      MANIFEST_JSON=$(echo $PIPELINE_MANIFEST | jq ".[] | select(.\"image-name\" | match(\"$COMPONENT\";\"i\"))")
      echo -e "Manifest Tag: $MANIFEST_JSON\n"

      # Extract the image tag from the manifest.
      TAG=$(echo $MANIFEST_JSON | jq -r ".\"image-tag\"")  
    else
      # Set the image tag to the default/choice image.
      TAG=$DEFAULT_SNAPSHOT

      if [[ "$COMPONENT" == "postgresql-13" ]]; then
        TAG=$DEFAULT_POSTGRESQL_SNAPSHOT 
      fi
    fi

    # Build the latest image tag that will be used within the bundle.
    LATEST_TAG=$IMAGE:$TAG

    log_color "cyan" "Build Image: $LATEST_TAG"
    update_images_csv $COMPONENT $LATEST_TAG
  fi
done

echo "Checking image within csv after update.."
get_images_from_csv

# Check to see if there are any changes made to the search-v2-operator csv file.
if [[ `git status --porcelain | grep $OPERATOR_CSV_FILEPATH` ]]; then
  update_doc_entry
fi

exit 0
